### PR TITLE
serializers: export for Bibtex

### PR DESCRIPTION
**Pull request by TLGINO**
* closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1208

### CODE REVIEW

File: invenio_rdm_records/resources/serializers/marcxml/schema.py [hunk 1]
```diff
@@ -37,20 +38,24 @@ class MARCXMLSchema(Schema):
...
     sizes = fields.Method("get_sizes")
     version = fields.Method("get_version")
     funding = fields.Method("get_funding")
 
-    def get_version(self, obj):
+    @staticmethod
+    def get_version(obj):
         """Get the version of the record."""
-        return obj["metadata"].get("version", missing)
+        version = obj.get("versions", {})
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/marcxml/schema.py:55]: I think this is the revision id (internal), while `metadata.version` is what we want to serialize? (i.e. version given in the deposit form by the user.

Comment (TLGINO) [reply invenio_rdm_records/resources/serializers/marcxml/schema.py:55]: That's what it was like before, for some reason I thought this was an error :sweat_smile: 
I'll put it back the way it was :)

File: invenio_rdm_records/resources/serializers/__init__.py [hunk 2]
```diff
@@ -34,6 +35,6 @@
     "IIIFManifestV2JSONSerializer",
     "IIIFSequenceV2JSONSerializer",
     "MARCXMLSerializer",
-    "StringCitationSerializer",
+    "BibtexSerializer",
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/__init__.py:38]: major: should add, not replace. Also, we dont enforce but it is nice if they are ordered alphabetically.

File: invenio_rdm_records/resources/serializers/bibtex/__init__.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""BibTex Serializer for Invenio RDM Records."""
+
+import json
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/bibtex/__init__.py:10]:
```suggestion

```

File: invenio_rdm_records/resources/serializers/bibtex/schema.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+class BibTexSchema(BaseSchema):
+    """Schema for records in BibTex."""
+
+    id = fields.Method("get_id")
+    resource_id = fields.Method("get_resource_id")
+    version = fields.Method("get_version")
+    date_created = fields.Method("get_date_created")
+    locations = fields.Method(
+        "get_locations"
+    )  # Corresponds to locations in the metadata schema
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/bibtex/schema.py:29]:
```suggestion
    ) 
```

File: invenio_rdm_records/resources/serializers/bibtex/schema.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+from slugify import slugify
+
+from ..schemas import BaseSchema
+from .schema_formats import BibTexFormatter
+
+
+class BibTexSchema(BaseSchema):
+    """Schema for records in BibTex."""
+
+    id = fields.Method("get_id")
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/bibtex/schema.py:23]: In this case, I'm not sure it needs a method field. Would it work by putting a SanitizedUnicode or Str (check for example what is used for id in the main `service/schema/`?

```suggestion
    id = fields.Str()
```

File: invenio_rdm_records/resources/serializers/bibtex/schema.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+            elif value is None and value in req_fields:
+                raise ValueError("Required value not found: {0}".format(value))
+
+        return out
+
+    def _format_output_row(self, field, value):
+        # [TODO] Refractor old zenodo code
+
+        out = ""
+        if isinstance(value, six.string_types):
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/bibtex/schema.py:146]: six was used for Python 2 and 3 compatibility, now we can replace this by `str` (remove the import too)

```suggestion
        if isinstance(value, str):
```

File: invenio_rdm_records/resources/serializers/bibtex/schema_formats.py [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+
+    def format(name, req_fields, opt_fields):
+        """Basic output formater for BibTexFormatter."""
+        return {
+            "name": name,
+            "req_fields": req_fields,
+            "opt_fields": opt_fields,
+        }
+
+    # def publication():
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/bibtex/schema_formats.py:57]: What is there todo with all the commented functions? should we open an issue to investigate?

File: invenio_rdm_records/resources/serializers/marcxml/__init__.py [hunk 8]
```diff
@@ -11,9 +11,10 @@
 import traceback
 from copy import deepcopy
 
+import flask_resources.serializers as bob
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/marcxml/__init__.py:14]:
```suggestion

```

File: invenio_rdm_records/resources/serializers/marcxml/schema.py [hunk 9]
```diff
@@ -33,25 +34,21 @@ class MARCXMLSchema(Schema):
...
         "get_types"
-    )  # Corresponds to ressource_type in the metadata schema
+    )  # Corresponds to resource_type in the metadata schema
     sources = fields.Method(
         "get_sources"
     )  # Corresponds to references in the metadata schema
-    coverage = fields.Method(
+    locations = fields.Method(
         "get_locations"
     )  # Corresponds to locations in the metadata schema
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/marcxml/schema.py:43]:
```suggestion
    ) 
```

File: invenio_rdm_records/resources/serializers/bibtex/schema_formats.py (hunk 7 above)
Comment (TLGINO) [reply invenio_rdm_records/resources/serializers/bibtex/schema_formats.py:57]: Yup there already is an open issue
https://github.com/inveniosoftware/invenio-rdm-records/issues/1231
as this is a cross-serializer problem

Comment (ppanero) [reply invenio_rdm_records/resources/serializers/bibtex/schema_formats.py:57]: Perfect, thanks! then we can take this in the new issue

**Comment by ppanero** [comment]
Waiting for CI to merge!

**Pull request merged**